### ISSUE TITLE
Target references and local paths are not evaluated in `ARG`s from the source Earthfile but in external referenced repo

### ISSUE THREAD

**Issue by Philipp-M** [earthly/earthly#2641]
Maybe I'm missing something, but I haven't found a way to pass files or evaluate local targets that are passed via an argument to a remote target reference.

I guess this example should explain the issue:

Host-repo:
```Earthfile
VERSION 0.6
FROM busybox
IMPORT github.com/user/earthly-lib:main

project:
    RUN echo "Hello world" > ./file
    SAVE ARTIFACT ./file

build:
    COPY --dir (earthly-lib+build/ --PROJECT=./ --PROJECT2=+project/file) ./build
```
earthly-lib:
```Earthfile
VERSION 0.6

build:
    FROM busybox
    ARG --required PROJECT
    ARG --required PROJECT2
    mkdir out
    COPY --dir ${PROJECT} ./out/
    COPY --dir ${PROJECT2} ./out/
```

It seems that the arguments (`PROJECT` and `PROJECT2`) are just replaced and tried to evaluate locally (inside the earthly-lib Earthfile context), so either the path `PROJECT=./` and the target reference in  `PROJECT2=+project/file` is wrongly evaluated (`+project` doesn't exist in the earthly-lib Earthfile).

Is it somehow possible to pass cross-refs and local paths into cross repo target references?

(I know there might be workarounds using `docker run` or UDCs, but I think the way above it would actually be cleaner.

**Comment by alexcb** [comment]
You're correct in noting that UDCs were created as a way to work around this issue, but it's a good point that they can't really be used with a COPY.

To use a UDC with a COPY, one would always have to have some boilerplate like `+project`, which creates (or copies in) the input, calls the UDC, then saves the output:

```
project:
    RUN echo "Hello world" > ./file
    DO +SOME-COMMON-UDC
    SAVE ARTIFACT ./file.out

build:
    COPY +project/file.out .
```

lib:

```
VERSION 0.6

SOME-COMMON-UDC:
    COMMAND
    ARG PATH
    RUN ./transform $PATH > $PATH.out
```

I agree this feels cumbersome, and could be improved.

#1892 proposes a `--self-contained` `COMMAND` modifier, which seems to overlap with this use-case.
